clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AvroComputeAggregationResponse.java

```diff
@@ -0,0 +1,51 @@
+package com.linkedin.venice.client.store;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import org.apache.avro.util.Utf8;
+
+
+/**
+ * Implementation of {@link ComputeAggregationResponse} that handles the results of count-by-value aggregations.
+ */
+public class AvroComputeAggregationResponse<K> implements ComputeAggregationResponse {
+  private final Map<K, ComputeGenericRecord> computeResults;
+  private final Map<String, Integer> fieldTopKMap;
+
+  public AvroComputeAggregationResponse(
+      Map<K, ComputeGenericRecord> computeResults,
+      Map<String, Integer> fieldTopKMap) {
+    this.computeResults = computeResults;
+    this.fieldTopKMap = fieldTopKMap;
+  }
+
+  @Override
+  public Map<String, Integer> getValueToCount(String field) {
+    Map<String, Integer> valueToCount = new HashMap<>();
+
+    for (ComputeGenericRecord record: computeResults.values()) {
+      Object value = record.get(field);
+      if (value instanceof Utf8) {
+        value = value.toString();
+      }
+      String key = (value == null) ? null : value.toString();
+      valueToCount.merge(key, 1, Integer::sum);
+    }
+
+    // Sort by count in descending order
+    Map<String, Integer> sortedMap = new LinkedHashMap<>();
+    valueToCount.entrySet()
+        .stream()
+        .sorted(Map.Entry.<String, Integer>comparingByValue().reversed())
+        .limit(fieldTopKMap.getOrDefault(field, Integer.MAX_VALUE))
+        .forEach(entry -> sortedMap.put(entry.getKey(), entry.getValue()));
+
+    return sortedMap;
```

Comment:
Here you just returned a Map of <String, Integer>, user normally expecting a certain type as interface provided:  `<T> Map<T, Integer> getValueToCount(String fieldName);`, you could explore `AvroGenericDeserializer` to see if we could deserialize by requested type. Here we only need to deserialize topK unique values.